### PR TITLE
adding dynamic instrumentation faq and add minor clarifications

### PR DESCRIPTION
**Pull request by shatzi**
### What does this PR do?
* Add FAQ to dynamic instrumentation as further reading.
* Fixed based on feedback from installing DI:
  * Allow options to configure remote-config using env variables
  * Explain RBAC required
  * Make creation of the log index optional if no indices with filter rules where defined. 

### Motivation
Improve docs based on feedback

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/dynamic_instrumentation/_index.md [hunk 1]
```diff
@@ -39,20 +43,41 @@ Dynamic Instrumentation requires the following:
 - [Unified Service Tagging][5] tags `service`, `env`, and `version` are applied to your deployment.
 - Optionally, [Source Code Integration][6] is set up for your service.
 
+**Note**: `debugger_read` and `debugger_write` permissions required to access Dynamic Instrumentation. While Datadog Administrator have those permissions by default, for more information about default and custom roles and assign roles to users, see [Role Based Access Control][11].
```
Comment (ojung) [inline content/en/dynamic_instrumentation/_index.md:46]:
```suggestion
**Note**: `debugger_read` and `debugger_write` permissions are required to access the Dynamic Instrumentation page. For more information about roles and on how to assign roles to users, see [Role Based Access Control][11].
```
Also, should we rename the roles? 😅

File: content/en/dynamic_instrumentation/_index.md [hunk 2]
```diff
@@ -39,20 +43,41 @@ Dynamic Instrumentation requires the following:
...
 
 1. Go to the [Remote Configuration setup page][7] and enable the feature for your organization.
 2. Create a key. 
 3. Update your `datadog-agent` with the provided configuration snippet.
-4. Set `remote_configuration.refresh_interval: 5s` in the `datadog-agent` configuration.
+4. Set `remote_configuration.refresh_interval` to be 5[sec].
+{{< tabs >}}
+{{% tab "Datadog-agent config.yaml" %}}
+
+Modify `datadog-agent` configuration:
```
Comment (ojung) [inline content/en/dynamic_instrumentation/_index.md:57]:
```suggestion
Modify `datadog-agent.yaml`:
```

File: content/en/dynamic_instrumentation/_index.md (hunk 1 above)
Comment (shatzi) [reply content/en/dynamic_instrumentation/_index.md:46]: I was thinking the same. but we decided to keep those when we rename the product - will sync with Evgeny to see if that changed.

File: content/en/dynamic_instrumentation/_index.md [hunk 3]
```diff
@@ -6,6 +6,10 @@ aliases:
 is_beta: true
 private: true
 further_reading:
+- link: "/dynamic_instrumentation/faq"
+  tag: "Documentation"
+  text: "Learn more about how Dynamic Instrumentation works"
```
[the remark this answers is not in the document]
Comment (shatzi) [reply content/en/dynamic_instrumentation/_index.md:11]: changed tag to FAQ?

File: content/en/dynamic_instrumentation/faq.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  - private: true
+---
+
+## What languages are supported by Dynamic Instrumentation?
+
+Java, Python and .NET
+
+<!-- {{< whatsnext desc="List of Frequently Asked Questions:" >}}
+    {{< nextlink href="/logs/faq/how-to-investigate-a-log-parsing-issue" >}}How to investigate a log parsing issue?{{< /nextlink >}}
+{{< /whatsnext >}} -->
```
[the remark this answers is not in the document]
Comment (shatzi) [reply content/en/dynamic_instrumentation/faq.md:14]: leftovers. removed

File: content/en/dynamic_instrumentation/faq.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+## How does Dynamic Instrumentation works?
+
+Dynamic instrumentation allows users to add probes to their production systems at anypoint in their code (and even 3rd party libraries). Dynamic Instrumentation has low overhead and is guaranteed to have no side effects on your system.
+
+## What data does Dynamic Instrumentation Snapshot probe captures
+
+Dynamic instrumentation capture the current context of a method or line that includes:
+* Method Arguments, Local Variables and fields. 
+* * Captures 3 levels deep of class objects
```
[the remark this answers is not in the document]
Comment (shatzi) [reply content/en/dynamic_instrumentation/faq.md:24]: line added

File: content/en/dynamic_instrumentation/faq.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+* * Limit to the first 100 items inside collections
+* * Limit string values to 255 characters
+* * Limit to 20 collected fields in an object
```
[the remark this answers is not in the document]
Comment (shatzi) [reply content/en/dynamic_instrumentation/faq.md:27]: fixed

File: content/en/dynamic_instrumentation/faq.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+* Call Stack
+* Caught & uncaught exceptions 
+
+## What probe types does Dynamic Instrumentation support?
+
+* *Snapshots probes* - captures the context of a method/line (rate limited for 1 snapshot per sec)
+* * Method arguments, Local Variables and fields, all captured at 3 levels deep.
+* * Stack trace
+* * Exceptions
+* *Metric probes* - generate a dynamic metric
```
[the remark this answers is not in the document]
Comment (shatzi) [reply content/en/dynamic_instrumentation/faq.md:37]: added

**Comment by hestonhoffman** [comment]
I've added [a ticket](https://datadoghq.atlassian.net/browse/DOCS-4396) to the docs queue for this one.

**Comment by kayayarai** [comment]
@shatzi @ojung I gave this PR a thorough edit, including overhauling the so-called FAQ page to fit more with our information architecture, to remove redundancies (such as listing the supported languages -- we try to minimize the number of places we need to update rote information like that every time a language is added), and to fix formatting and grammar. Please let me know if you have questions. If you're happy with it, let me know when we can ship it.

**Review by kayayarai — APPROVED** [review]

**Review by ojung — APPROVED** [review]
I like the changes, thanks @kayayarai 🙇‍♂️

**Pull request merged**